File: android/core/data/src/main/java/com/akaar/core/data/remote/SupabaseConfig.kt
package com.akaar.core.data.remote

/**
 * Supabase connection settings.
 *
 * Only the publishable anon key is ever present in the app. It is not a secret
 * in the usual sense and it grants nothing on its own: row level security and
 * the SECURITY DEFINER functions are the actual boundary, which is why a
 * decompiled APK still cannot read a seller's costs or accept an offer.
 *
 * No AI provider key ever reaches the device. Those live in Edge Function
 * secrets, server side.
 */
data class SupabaseConfig(
    val url: String,
    val anonKey: String,
) {
    val isConfigured: Boolean get() = url.isNotBlank() && anonKey.isNotBlank()

    companion object {
        /** RPC names, matching supabase/migrations/009_functions.sql. */
        object Rpc {
            const val RECALCULATE_PRICE = "recalculate_price"
            const val SET_PRICE_CORRIDOR = "set_price_corridor"
            const val PUBLISH_PRODUCT = "publish_product"
            const val PLACE_OFFER = "place_offer"
            const val RESPOND_TO_OFFER = "respond_to_offer"
            const val BUYER_RESPOND_OFFER = "buyer_respond_offer"
            const val SEARCH_MARKETPLACE = "search_marketplace"
        }

        /** Tables a client may read. Writes with a business rule go through Rpc. */
        object Table {
            const val PROFILES = "profiles"
            const val ARTISAN_PROFILES = "artisan_profiles"
            const val ARTISAN_PRIVATE = "artisan_private"
            const val BUYER_PROFILES = "buyer_profiles"
            const val PRODUCTS = "products"
            const val PRODUCT_PRICING_PRIVATE = "product_pricing_private"
            const val PRODUCT_MEDIA = "product_media"
            const val CRAFTS = "crafts"
            const val MATERIALS = "materials"
            const val TECHNIQUES = "techniques"
            const val OFFERS = "offers"
            const val CUSTOM_REQUESTS = "custom_requests"
            const val ORDERS = "orders"
            const val CONVERSATIONS = "conversations"
            const val MESSAGES = "messages"
            const val NOTIFICATIONS = "notifications"
        }
    }
}

File: mobile/apps/mobile/src/data/supabaseConfig.ts
/**
 * Port of android/core/data/.../SupabaseConfig.kt.
 *
 * Only the publishable anon key is ever present in the app. It is not a
 * secret in the usual sense and it grants nothing on its own: row level
 * security and the SECURITY DEFINER functions are the actual boundary,
 * which is why a decompiled/bundled app still cannot read a seller's costs
 * or accept an offer.
 *
 * No AI provider key ever reaches the device. Those live in Edge Function
 * secrets, server side. Values are injected via react-native-config /
 * .env at build time - never committed - mirroring how the Kotlin side
 * pulls them from local.properties / CI secrets into BuildConfig.
 */
export interface SupabaseConfig {
  url: string;
  anonKey: string;
}

export function isConfigured(config: SupabaseConfig): boolean {
  return config.url.trim().length > 0 && config.anonKey.trim().length > 0;
}

/** RPC names, matching supabase/migrations/*_functions.sql. */
export const SupabaseRpc = {
  RecalculatePrice: "recalculate_price",
  SetPriceCorridor: "set_price_corridor",
  PublishProduct: "publish_product",
  PlaceOffer: "place_offer",
  RespondToOffer: "respond_to_offer",
  BuyerRespondOffer: "buyer_respond_offer",
  SearchMarketplace: "search_marketplace",
} as const;

/** Tables a client may read. Writes with a business rule go through Rpc. */
export const SupabaseTable = {
  Profiles: "profiles",
  ArtisanProfiles: "artisan_profiles",
  ArtisanPrivate: "artisan_private",
  BuyerProfiles: "buyer_profiles",
  Products: "products",
  ProductPricingPrivate: "product_pricing_private",
  ProductMedia: "product_media",
  Crafts: "crafts",
  Materials: "materials",
  Techniques: "techniques",
  Offers: "offers",
  CustomRequests: "custom_requests",
  Orders: "orders",
  Conversations: "conversations",
  Messages: "messages",
  Notifications: "notifications",
} as const;
